### DISCUSSION TITLE
Commercial use - onnxruntime with CUDA

### DISCUSSION THREAD

**Discussion by mcicat** [microsoft/onnxruntime#8689] [Other Q&A]
Hello everyone, 
if I would like to use onnxruntime for inferencing a model on a customer PC with cuda is it required to install Cuda Toolkit on that machine? (I initially thought that Cuda runtimes are included with the Onnxruntime). 

If yes, that means that Visual Studio should also be installed on that machine. Is it legally okay to install the Visual Studio Community for this purpose? Community version is available for single developer projects in case of commercial use, but in this case it won't be used for development at all. It's just needed for the libraries...

**Comment by ytaous** [comment]
@faxu

**Reply by faxu** [reply]
CUDA is required to be installed and is not directly distributed with the ORT package. The ORT package does contain the [execution provider](https://onnxruntime.ai/docs/reference/execution-providers/) that interfaces with CUDA. 

Visual Studio does not need to be installed, as long as the version matches what's required by the ORT version - https://onnxruntime.ai/docs/reference/execution-providers/CUDA-ExecutionProvider.html#requirements. CUDA can be installed on NVIDIA's site: https://developer.nvidia.com/cuda-toolkit-archive

**Reply by mcicat** [reply]
Visual Studio is a requirement for installing Cuda (as stated here https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/index.html). Are you trying to say, that ORT will work if I install Cuda without VS (it is possible, but Cuda complains and notifies you, that it probably won't work), or that VS is not needed by the ORT itself?

**Reply by HectorSVC** [reply]
CUDA is a toolkit to help develop platform/application on Nvidia GPU, it relies on VS to build the code.  If you just run Ort with CUDA , there's no need to have VS installed. Application developer's can have the required Ort lib and CUDA runtime libs packed into their package so that end user don't need to install it.

**Reply by mcicat** [reply]
Okay, I think I'm getting it. But to sum it up, just one last question. Will inference with Cuda work if I install Cuda without Visual Studio even tough VS is a requirement for Cuda?

**Reply by HectorSVC** [reply]
Yes. It will work.

**Reply by mcicat** [reply]
Thank You very much.
